pkg/provider/aws/expose.go

```diff
@@ -0,0 +1,123 @@
+		routeTrigger := fmt.Sprintf("%s:%s:%s", route.ExecUnitName, route.Path, route.Verb)
+		triggers[routeTrigger] = routeTrigger
+
+		segments := strings.Split(route.Path, "/")
+		currPathSegment := strings.Builder{}
+
+		for _, segment := range segments {
+			methodRequestParams := map[string]bool{}
+			integrationRequestParams := map[string]string{}
+
+			if strings.Contains(segment, ":") {
+				pathParam := fmt.Sprintf("request.path.%s", segment)
+				pathParam = strings.ReplaceAll(pathParam, ":", "")
+				pathParam = strings.ReplaceAll(pathParam, "*", "")
+				methodRequestParams[fmt.Sprintf("method.%s", pathParam)] = true
+				integrationRequestParams[fmt.Sprintf("integration.%s", pathParam)] = fmt.Sprintf("method.%s", pathParam)
+			}
+
+			segment = convertPath(segment)
+			currPathSegment.WriteString(fmt.Sprintf("%s/", segment))
+			refs := []core.AnnotationKey{gateway.Provenance(), execUnit.Provenance()}
+			resource, ok := resourceMap[segment]
+			if !ok {
+				resource = resources.NewApiResource(api, refs, segment)
+				dag.AddResource(resource)
+				dag.AddDependency2(resource, api)
+				resourceMap[currPathSegment.String()] = resource
+				triggers[resource.Name] = resource.Name
+			}
+
+			method := resources.NewApiMethod(resource, refs, strings.ToUpper(string(route.Verb)), methodRequestParams)
+			dag.AddResource(method)
+			dag.AddDependency2(method, resource)
+			integration, err := a.createIntegration(method, execUnit, refs, route, dag)
+			if err != nil {
+				errs.Append(err)
+				continue
+			}
+
+			triggers[integration.Name] = integration.Name
+		}
+	}
+	api.ConstructsRef = api_references
+	deployment := resources.NewApiDeployment(api, api_references, triggers)
+	dag.AddResource(deployment)
+	dag.AddDependency2(deployment, api)
+	stage := resources.NewApiStage(deployment, "$default", api_references)
+	dag.AddResource(stage)
+	dag.AddDependency2(stage, deployment)
+	return errs.ErrOrNil()
+}
+
+func (a *AWS) createIntegration(method *resources.ApiMethod, unit *core.ExecutionUnit, refs []core.AnnotationKey, route core.Route, dag *core.ResourceGraph) (*resources.ApiIntegration, error) {
+	cfg := a.Config.GetExecutionUnit(unit.ID)
+	switch cfg.Type {
+	case Lambda:
+		constructResources, _ := a.GetResourcesDirectlyTiedToConstruct(unit)
+		if len(constructResources) != 1 {
+			return nil, errors.Errorf("Expected one resource to be tied to a lambda execution unit, %s, but found %s", unit.ID, strconv.Itoa(len(constructResources)))
+		}
+		function, ok := constructResources[0].(*resources.LambdaFunction)
+		if !ok {
+			return nil, errors.Errorf("Expected resource to be of type, lambda function, for execution unit, %s", unit.ID)
+		}
+		lambdaPermission := resources.NewLambdaPermission(function, core.IaCValue{Resource: method.RestApi, Property: core.ARN_IAC_VALUE}, "apigateway.amazonaws.com", "lambda:InvokeFunction", refs)
+		dag.AddResource(lambdaPermission)
+		dag.AddDependency2(lambdaPermission, function)
```

Comment:
Huh, am I double-confused? I thought this would be in the reverse order. In the visualization, it would look something like this, I think (right?):

```
GET /foo ➤ APIG ➤ Lambda ➤ Permission
```

Is the `AddDependency2` order the reverse of the arrows? I thought it used to be reverse, but now it's the same.

Reply:
yeah i was trying to say this, we go against the order of the visualizer now. This is the whole talk we had late last week

Reply:
Lol, okay I was so lost apparently. I thought we previously went against the order of the visualizer, and with `AddDependency2` we're in sync with it. Why are graphs so hard, haha.

Reply:
nope, opposite, but at this point i dont think we switch back now but im confused looking at our graphs now